pyhealth/datasets/base_dataset.py

```diff
@@ -593,3 +603,170 @@ def task_stat(self) -> str:
     def info():
         """Prints the output format."""
         print(INFO_MSG)
+
+class SampleDataset(ABC, Dataset):
+    """Abstract sample dataset class.
+
+    This dataset takes the processed data samples as an input list
+    
+    Args:
+        samples: the processed data samples.
+            E.g., 
+                samples[0] = {
+                    visit_id: 1,
+                    patient_id: 1,
+                    "codnition": ["A", "B"],
+                    "procedure": ["C", "D"],
+                    "label": 1
+                }
```

Comment:
Can you elaborate on the format of the samples? I think you can move the description in Attributes/samples here.